pkg/querier/queryrange/limits.go

```diff
@@ -509,6 +513,8 @@ func (rt limitedRoundTripper) RoundTrip(r *http.Request) (*http.Response, error)
 
 	sem := semaphore.NewWeighted(int64(parallelism))
 
+	start := time.Now()
```

Comment:
This needs to move before the semaphore is acquired on line 527. The handler is called multiple times, so we can't create a reference point beforehand because it'll seem like all subsequent requests are taking longer and longer.

Reply:
@dannykopping  please check the latest two commits where I wrapped the Aquire and the timing stuff in one method. It's primarily for the benefit of test.  There should be some elegant approach to do this , please feel free to point it out. Thanks